### PR TITLE
Add convenience methods for implementing `Endpoint`s

### PR DESCRIPTION
**Pull request by 987Nabil**

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala [hunk 1]
```diff
@@ -144,6 +144,24 @@ final case class Endpoint[Input, Err, Output, Middleware <: EndpointMiddleware](
...
+  def implementWith[Env](f: Input => Output): Routes[Env, Err, Middleware] =
+    implement(in => ZIO.succeed(f(in)))
+
+  /**
+   * Converts this endpoint, which is an abstract description of an endpoint,
+   * into a path, which maps a path to a handler for that path. In order to
+   * convert an endpoint into a path, you must specify the output, while the
+   * input is being ignored.
+   */
+  def implementSucceed[Env](f: => Output): Routes[Env, Err, Middleware] =
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala]:
```suggestion
  def implementAs[Env](f: => Output): Routes[Env, Err, Middleware] =
```

File: zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala [hunk 2]
```diff
@@ -144,6 +144,24 @@ final case class Endpoint[Input, Err, Output, Middleware <: EndpointMiddleware](
   def implement[Env](f: Input => ZIO[Env, Err, Output]): Routes[Env, Err, Middleware] =
     Routes.Single[Env, Err, Input, Output, Middleware](self, f)
 
+  /**
+   * Converts this endpoint, which is an abstract description of an endpoint,
+   * into a path, which maps a path to a handler for that path. In order to
+   * convert an endpoint into a path, you must specify a function which handles
+   * the input, and returns the output.
+   */
+  def implementWith[Env](f: Input => Output): Routes[Env, Err, Middleware] =
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala]:
```suggestion
  def implementPurely[Env](f: Input => Output): Routes[Env, Err, Middleware] =
```

**Comment by 987Nabil** [comment]
@jdegoes I also added `implementAsError`

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
Thank you!

**Pull request merged**